src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -89,12 +93,20 @@ private void addReject(long val) {
      * @param excludeValues   The terms to be excluded
      *                  (may only be {@code null} if one of the other arguments is none-null.
      */
-    public IncludeExclude(Pattern include, Pattern exclude, Set<BytesRef> includeValues, Set<BytesRef> excludeValues) {
-        assert includeValues != null || include != null || 
+    public IncludeExclude(RegExp include, RegExp exclude, Set<BytesRef> includeValues, Set<BytesRef> excludeValues) {
```

Comment:
what is this includeValues/excludeValues? Should it just build the same matcher too (maybe require SortedSet).